.github/workflows/ci.yaml

```diff
@@ -33,13 +46,112 @@ jobs:
         key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
         restore-keys: |
           ${{ runner.os }}-go-
+    - name: Run controller tests
+      run: make test-envtest
+  scorecard-test:
+    runs-on: ubuntu-latest
+    if: ${{ github.repository_owner == 'cryostatio' }}
+    env: 
+      SCORECARD_NAMESPACE: cryostat-operator-scorecard
+      SCORECARD_SA_NAME: cryostat-scorecard
+      GHCR_USERNAME: ${{ github.repository_owner }}
+      GHCR_PASSWORD: ${{ secrets.GHCR_PR_TOKEN }}
+      GHCR_SERVER: ghcr.io
+    steps:
+    - name: Fail if safe-to-test label NOT applied
+      if: ${{ github.event_name == 'pull_request_target' && !contains(github.event.pull_request.labels.*.name, 'safe-to-test') }}
+      run: exit 1
+    - uses: actions/checkout@v2
     - uses: jpkrohling/setup-operator-sdk@v1.1.0
       with:
         operator-sdk-version: v1.28.0
-    - run: make generate manifests manager test-envtest
-      if: ${{ github.event_name == 'pull_request' }}
-    - run: make oci-build
-      if: ${{ github.event_name == 'push' }}
+    - name: Build scorecard image for test
+      id: build-scorecard
+      run: |
+        make scorecard-build CUSTOM_SCORECARD_IMG=ghcr.io/${{ github.repository_owner }}/cryostat-operator-scorecard:ci-$GITHUB_SHA
+        echo "tag=ci-$GITHUB_SHA" >> $GITHUB_OUTPUT
+    - name: Push scorecard image to ghcr.io for test
+      id: push-scorecard-to-ghcr
+      uses: redhat-actions/push-to-registry@v2
+      with:
+        image: cryostat-operator-scorecard
+        tags: ${{ steps.build-scorecard.outputs.tag }}
+        registry: ghcr.io/${{ github.repository_owner }}
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GHCR_PR_TOKEN }}
+    - name: Build bundle image for test
+      id: build-bundle
+      run: |
+        make bundle bundle-build BUNDLE_IMG=ghcr.io/${{ github.repository_owner }}/cryostat-operator-bundle:ci-$GITHUB_SHA
+        echo "tag=ci-$GITHUB_SHA" >> $GITHUB_OUTPUT
+    - name: Push bundle image to ghcr.io for test
+      id: push-bundle-to-ghcr
+      uses: redhat-actions/push-to-registry@v2
+      with:
+        image: cryostat-operator-bundle
+        tags: ${{ steps.build-bundle.outputs.tag }}
+        registry: ghcr.io/${{ github.repository_owner }}
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GHCR_PR_TOKEN }}
+    - name: Set up Kind cluster
+      run: |
+        {
+            printf -- "kind: Cluster\n"
+            printf -- "apiVersion: kind.x-k8s.io/v1alpha4\n"
+            printf -- "nodes:\n"
+            printf -- "- role: control-plane\n"
+            printf -- "  kubeadmConfigPatches:\n"
+            printf -- "  - |\n"
+            printf -- "    kind: InitConfiguration\n"
+            printf -- "    nodeRegistration:\n"
+            printf -- "      kubeletExtraArgs:\n"
+            printf -- "        node-labels: \"ingress-ready=true\"\n"
+        } > kind-config.yaml
```

Comment:
I can't find any concrete docs examples but I think this can just be a file at `.github/workflows/kind-config.yaml` for example, and then `kind create cluster` can just reference the file path from the filesystem. I'm not sure what the working directory of the process is so you might need to give it a relative path to the right directory where that file is.

Reply:
I think `actions/checkout@v2` should set the current working dir to our repo. Then, it should just be `.github/workflows/kind-config.yaml`.

Reply:
I suppose the config could be useful for other testing with `kind`, including CI run elsewhere than GitHub. A top-level `scripts` or `testing` directory in the repo would make sense to me. @ebaron any opinion?

Reply:
For now I'm leaning towards under `.github/`, Cryostat has similar CI-related things there: https://github.com/cryostatio/cryostat/tree/main/.github